src/main/java/com/powsybl/nad/build/iidm/NetworkGraphBuilder.java

```diff
@@ -70,119 +72,125 @@ private VoltageLevelNode createVoltageLevelNode(VoltageLevel vl, Graph graph, bo
         return vlNode;
     }
 
-    private class VisitorBuilder extends DefaultTopologyVisitor {
-        private final Graph graph;
+    private void visitEquipments(VoltageLevel vl, Graph graph) {
+        vl.getLineStream().forEach(l -> visitLine(vl, l, graph));
+        vl.getTwoWindingsTransformerStream().forEach(twt -> visitTwoWindingsTransformer(vl, twt, graph));
+        vl.getThreeWindingsTransformerStream().forEach(thwt -> visitThreeWindingsTransformer(vl, thwt, graph));
+        vl.getConnectableStream().filter(HvdcConverterStation.class::isInstance).forEach(hvdc -> visitHvdcConverterStation((HvdcConverterStation<?>) hvdc, graph));
```

Comment:
here you need to call `vl.getConnectableStream(HvdcConverterStation.class)` instead. Indeed, `vl.getConnectableStream()` will load all the connectables from the the database, even all those which we don't need.